popover-api/popover-hint/index.css

```diff
@@ -0,0 +1,112 @@
+/* General styling */
+
+* {
+  box-sizing: border-box;
+}
+
+html {
+  font-family: sans-serif;
+  height: 100%;
+}
+
+body {
+  height: inherit;
+  background: #ccc;
+  margin: 0;
+}
+
+#wrapper {
+  height: inherit;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+/* Button bar styling */
+
+#button-bar {
+  height: 48px;
+  width: 30%;
+  min-width: 300px;
+  padding: 8px;
+  border-radius: 6px;
+  background: #eee;
+  box-shadow: 1px 1px 3px #999;
+  display: flex;
+  justify-content: space-around;
+  gap: 10px;
+}
+
+/* Button styling */
+
+#button-bar button {
+  border: 1px solid #999;
+  border-radius: 4px;
+  background: #eee;
+  flex: 1;
+}
+
+#button-bar button:hover,
+#button-bar button:focus {
+  background: #ccc;
+}
+
+#button-bar button:active {
+  box-shadow: inset 1px 1px 2px #333, inset -1px -1px 2px #eee;
+}
+
+/* Styling auto popovers */
+
+[popover="auto"] {
+  inset: unset;
+  position: absolute;
+  bottom: calc(anchor(top) + 20px);
+  justify-self: anchor-center;
+  margin: 0;
+
+  width: 200px;
+  text-align: center;
+  padding: 8px;
+  border-radius: 6px;
+  background: #eee;
+  box-shadow: 1px 1px 3px #999;
+  border: none;
+}
+
+[popover="auto"] button {
+  border: 1px solid #999;
+  border-radius: 4px;
+  background: #eee;
+  width: 100%;
+  line-height: 1.5;
+}
+
+[popover="auto"] button:first-of-type {
+  margin-bottom: 8px;
+}
+
+[popover="auto"] button:hover,
+[popover="auto"] button:focus {
+  background: #ccc;
+}
+
+[popover="auto"] button:active {
+  box-shadow: inset 1px 1px 2px #333, inset -1px -1px 2px #eee;
+}
+
+/* Styling hint popovers */
+
+[popover="hint"] {
+  inset: unset;
+  position: absolute;
+  top: calc(anchor(bottom) + 5px);
```

Comment:
Only issue I have with this is that it looks weird on browsers that don't support it, you get a tooltip off the bottom of the page and some scrollbars on mouseover. It looks like it's hard to do proper feature detection on that level of granularity (specifically ~`supports popover = hint`), isn't it?

Reply:
You could feature detect based on the popover property of the DOM element, but it'd be a right pain building an alternative that still provides an acceptable experience in non-supporting browsers. What do think is needed here?

Reply:
OK, I've added in a note about browser support, and added in a media query to stop the note box from covering the button bar on narrow screens